### PR TITLE
Part 2 changes

### PR DESCRIPTION
**Pull request by dyazdani**
Changes made for part two of this project

### CODE REVIEW

File: __tests__/costumes.test.js [hunk 1]
```diff
@@ -8,6 +8,18 @@ const {
     getPool
 } = require(".././index");
 
+const { 
+    matchesDatabase,
```
Comment (paloobi) [inline __tests__/costumes.test.js]: Not: as a general rule, in test utils, I recommend using functions that return the test data, that way they can be reused across multiple test files without reusing the same object. It's best if each test uses fresh data to ensure it's not possible to modify the same object in multiple tests

File: __tests__/costumes.test.js [hunk 2]
```diff
@@ -153,7 +73,7 @@ describe("createCostume adapter", () => {
             SELECT * FROM costumes WHERE name='buttless chaps';
         `);
         
-        expect(matchesCostumeInDatabase(buttlessChaps, chapsFromDatabase)).toBe(true);
+        expect(matchesDatabase(buttlessChaps, chapsFromDatabase)).toBe(true);
```
Comment (paloobi) [inline __tests__/costumes.test.js]: Avoid assertions in after all - assertions should happen in each test

**Review by paloobi — COMMENTED** [review]
Adding a couple comments, will review more later (currently out and about)

File: __tests__/costumes.test.js (hunk 1 above)
Comment (dyazdani) [reply __tests__/costumes.test.js]: So, import functions like this instead?: 
```
const getBallroomGown = () => {
    const ballroomGown = {
        name: "ballroom gown",
        category: "adult",
        gender: "female",
        size: "L",
        type: "dress",
        stockCount: 1,
        price: 150.99
    };
    return ballroomGown;
}

File: __tests__/costumes.test.js (hunk 2 above)
Comment (dyazdani) [reply __tests__/costumes.test.js]: Do you mean in an afterAll call? I am not sure which assertions you are referring to. This is the code I have for the afterAll call (lines 32-35):
```
// Disconnect from postgres database after all tests done
afterAll(async () => {
    await pool.end()
})

**Comment by dyazdani** [comment]
Question: is it OK to use other adapters in my tests besides the one I am testing? I use `createTables` a lot, as well as others, to keep the test code from too long and to keep it DRY.

File: utilities.js
Comment (paloobi) [inline utilities.js:1]: Test utils should live under `__tests__`

File: costumes.js [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    return costumes;
+}
+
+const getCostumeById = async (pool, id) => {
+    const {rows:[costume]} = await pool.query(`
+        SELECT *
+        FROM costumes
+        WHERE id = $1;
+    `, [id])
+    if (costume === undefined) {
```
Comment (paloobi) [inline costumes.js]: Will it equal `undefined` or `null`? FWIW I think it would be reasonable to just do a truthy check in this kind of situation (since null or undefined would both be unexpected states)
```suggestion
    if (!costume) {
```

File: costumes.js [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+        name, 
+        category, 
+        gender, 
+        size, 
+        type, 
+        stockCount, 
+        price, 
+        id
+    ])
+    if (costume === undefined) {
```
Comment (paloobi) [inline costumes.js]: same suggestion
```suggestion
    if (!costume) {
```

File: costumes.js [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+
+const deleteCostumeById = async (pool, id) => {
+    const {rows: costumes} = await pool.query(`
+        SELECT * FROM costumes 
+        WHERE id = $1;
+    `, [id])
+    
+    if (costumes.length === 0) {
+        throw new Error(`Could not delete row because id provided (${id}) does not exist in table.`)
+    }
```
Comment (paloobi) [inline costumes.js]: Is this necessary? What will happen if you try to delete a costume that doesn't exist?

File: customers.js [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+}
+
+const getAllCustomers = async (pool) => {
+    const {rows: customers} = await pool.query(`
+        SELECT * FROM customers;
+    `)
+    return customers;
+}
+
+const getCustomerLinkedToOrder = async (pool, orderId) => {
```
Comment (paloobi) [inline customers.js]: nit
```suggestion
const getCustomerByOrderID = async (pool, orderId) => {
```

File: customers.js [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+        WHERE id = $1;
+    `, [orderId])
+    if (order === undefined) {
+        throw new Error(`Could not retrieve data because id provided (${id}) does not exist in table.`)
+    } 
+
+    const {rows:[customer]} = await pool.query(`
+        SELECT * FROM customers
+        WHERE id = $1;
+    `, [order["customer_id"]])
```
Comment (paloobi) [inline customers.js]: Fairly certain you could achieve this without 2 separate queries if you use a `JOIN`

File: customers.js [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+    const {rows: [customer]} = await pool.query(`
+        UPDATE customers
+        SET 
+            full_name = $1,
+            email = $2,
+            password = $3
+        WHERE id = $4
+        RETURNING *;
+    `, [fullName, email, password, id])
+    if (customer === undefined) {
```
Comment (paloobi) [inline customers.js]:
```suggestion
    if (!customer) {
```

File: index.js [hunk 9]
```diff
@@ -27,9 +27,18 @@ const getPool = () => {
...
-        DROP TABLE IF EXISTS costumes;    
+        DROP TABLE IF EXISTS orders_costumes;
+        DROP TABLE IF EXISTS orders;  
+        DROP TABLE IF EXISTS customers;
+        DROP TABLE IF EXISTS costumes; 
     `)
     await pool.query(` 
+    CREATE TABLE customers(
+        id SERIAL PRIMARY KEY,
+        full_name VARCHAR(80) NOT NULL,
```
Comment (paloobi) [inline index.js]: I think this may be too few characters for fullname? esp. when you take into account i18n.

I think it's standard practice to just use 255 for most varchars (that way you can enforce stricter limits without them being locked in stone)

File: index.js [hunk 10]
```diff
@@ -46,130 +55,36 @@ const createTables = async (pool) => {
...
             size VARCHAR(10) NOT NULL,
             type VARCHAR(80) NOT NULL,
-            stock_count INTEGER NOT NULL,
+            stock_count INT NOT NULL,
             price FLOAT NOT NULL 
         );
+        CREATE TABLE orders(
+            id SERIAL PRIMARY KEY,
+            date_placed DATE NOT NULL,
+            status VARCHAR NOT NULL CHECK(status IN(
```
Comment (paloobi) [inline index.js]:
```suggestion
            status VARCHAR(255) NOT NULL CHECK(status IN(
```

File: index.js [hunk 11]
```diff
@@ -46,130 +55,36 @@ const createTables = async (pool) => {
...
+        );
+        CREATE TABLE orders_costumes(
+            id SERIAL PRIMARY KEY,
+            order_id INT REFERENCES orders(id),
+            costume_id INT REFERENCES costumes(id)
+        );
     `)
 }
 
-const createCostume = async (
```
Comment (paloobi) [inline index.js:55]: heads up - for code changes that move code around, I recommend a separate PR for refactor and for the new features. It makes it easier to review code. It's not a big deal, just for future reference

File: orders.js [hunk 12]
```diff
@@ -0,0 +1,89 @@
...
+    `)
+    return orders;
+}
+
+const getOrderById = async (pool, id) => {
+    const {rows:[order]} = await pool.query(`
+        SELECT * FROM orders
+        WHERE id = $1;
+    `, [id])
+    if (order === undefined) {
```
Comment (paloobi) [inline orders.js]:
```suggestion
    if (!order) {
```

File: orders.js [hunk 13]
```diff
@@ -0,0 +1,89 @@
...
+        throw new Error(`Could not retrieve data because id provided (${id}) does not exist in table.`)
+    } 
+    return order;
+}
+
+const getCustomerOrders = async (pool, customerId) => {
```
Comment (paloobi) [inline orders.js]:
```suggestion
const getOrdersByCustomerId = async (pool, customerId) => {
```

Comment (paloobi) [reply orders.js]: This is especially important because it could lead to confusion and make someone think you are referring to a join table

File: orders.js [hunk 14]
```diff
@@ -0,0 +1,89 @@
...
+        WHERE id = $1;
+    `, [customerId])
+    if (customer === undefined) {
+        throw new Error(`Could not retrieve data because id provided (${id}) does not exist in table.`)
+    } 
+
+    const {rows: orders} = await pool.query(`
+        SELECT * FROM orders
+        WHERE customer_id = $1;
+    `, [customer.id])
```
Comment (paloobi) [inline orders.js]: Use `JOIN`

File: ordersCostumes.js [hunk 15]
```diff
@@ -0,0 +1,117 @@
...
+    } 
+
+    // throw error if order id does not exist
+    const {rows:[order]} = await pool.query(`
+        SELECT * FROM orders
+        WHERE id = $1;
+    `, [orderId])
+    if (order === undefined) {
+        throw new Error(`Could not retrieve data because id provided for order (${id}) does not exist in table.`)
+    }
```
Comment (paloobi) [inline ordersCostumes.js]: Don't think this is necessary? I'm pretty sure it will throw an error if you call INSERT and don't pass both IDs. If it doesn't throw an error - make sure these are not allowed to be null in table definition

File: ordersCostumes.js [hunk 16]
```diff
@@ -0,0 +1,117 @@
...
```
Comment (paloobi) [inline ordersCostumes.js]: Same, this shouldn't be necessary, won't SQL error if you pass invalid IDs or no IDs?

Comment (paloobi) [reply ordersCostumes.js]: Also this is repeat code - it would be better to check this kind of thing not at the DB adapter layer (eg. in the web server, you might re-use getOrder and getCostume first to check)

File: ordersCostumes.js [hunk 17]
```diff
@@ -0,0 +1,117 @@
...
+    } 
+    
+    // remove costume
+    await pool.query(`
+        DELETE FROM orders_costumes
+        WHERE costume_id = $1 AND order_id = $2;
+    `, [costumeId, orderId]);
+}
+
+const getAllOrdersOfCostumeById = async (pool, costumeId) => {
```
Comment (paloobi) [inline ordersCostumes.js:28]: I'm not sure if this is needed? What is the feature you're thinking of that would require this?
```suggestion
const getOrdersContainingCostumeId = async (pool, costumeId) => {
```

File: ordersCostumes.js [hunk 18]
```diff
@@ -0,0 +1,117 @@
...
+    // throw error if costume id does not exist
+    const {rows:[costume]} = await pool.query(`
+        SELECT * FROM costumes
+        WHERE id = $1;
+    `, [costumeId])
+    if (costume === undefined) {
+        throw new Error(`Could not retrieve data because id provided for costume (${id}) does not exist in table.`)
+    }
```
Comment (paloobi) [inline ordersCostumes.js]: again, don't recommend doing this kind of error check at the DB layer

File: ordersCostumes.js [hunk 19]
```diff
@@ -0,0 +1,117 @@
...
+    FROM orders_costumes 
+    JOIN orders 
+    ON order_id = orders.id 
+    WHERE costume_id=$1;
+    `, [costumeId]
+    )
+    return orders;
+}
+
+const getAllCostumesFromOrderById = async (pool, orderId) => {
```
Comment (paloobi) [inline ordersCostumes.js]:
```suggestion
const getCostumesByOrderId = async (pool, orderId) => {
```

File: ordersCostumes.js [hunk 20]
```diff
@@ -0,0 +1,117 @@
...
+    // throw error if order id does not exist
+    const {rows:[order]} = await pool.query(`
+        SELECT * FROM orders
+        WHERE id = $1;
+    `, [orderId])
+    if (order === undefined) {
+        throw new Error(`Could not retrieve data because id provided for order (${id}) does not exist in table.`)
+    }
```
Comment (paloobi) [inline ordersCostumes.js]: same - not sure if you want to do this here, or in web server layer - but I think it's fine either way

File: seed/seedData.json [hunk 21]
```diff
@@ -0,0 +1,49 @@
...
+                "status": "pending",
+                "customerId": 1
+            },
+            {
+                "datePlaced": "2020-05-23 13:33:12-06",
+                "status": "completed",
+                "customerId": 2
+            }
+        ]
+}
```
Comment (paloobi) [inline seed/seedData.json:49]: Recommend also adding some `orders_costumes` so you can add some costumes to orders in the seed file

File: __tests__/costumes.test.js (hunk 2 above)
Comment (paloobi) [reply __tests__/costumes.test.js]: Oh sorry - GitHub was hiding some lines, so it looked like it had an `expect` in the `afterAll`

File: utilities.js [hunk 22]
```diff
@@ -0,0 +1,317 @@
+// *** FOR JEST TESTING ***
+
+// Helper function for matching input objects with objects returned from DB queries
+const matchesDatabase = (inputObject, databaseObject) => {
+    let areAllPropertiesMatched = true;
```
Comment (paloobi) [inline utilities.js]: this is a little concerning - prefer to assume they don't match and then determine if they do. Defaulting to true could lead to some scary bugs that would render all of your tests inaccurate

File: utilities.js [hunk 23]
```diff
@@ -0,0 +1,317 @@
...
+                areAllPropertiesMatched = false;
+                return areAllPropertiesMatched;
+            }
+
+        if (!(databaseObject[propForDatabaseObject] instanceof Date) &&
+            inputObject[prop] !== databaseObject[propForDatabaseObject]) {
+            areAllPropertiesMatched = false;
+            return areAllPropertiesMatched;
+        }
+    }
```
Comment (paloobi) [inline utilities.js]: Have you looked into whether `jest` already built something like this for checking if an object matches another object?

**Review by paloobi — CHANGES_REQUESTED** [review]
I think it would be worth talking more about your tests - the utils seem to be over-optimized to the point of making all the tests co-dependent on one function - I think you may want to rethink the strategy here. It is more important that each test purely test a given functionality.

The main changes I'm requesting:
- use JOIN in the couple of places I mentioned, instead of separately making 2 queries
- confirm that it's necessary to load before attempting DELETE 
- basically, overall, look for DB adapters where you're making 2 queries and once, and see if you can find a way to turn it into one query or if the extra queries are necessary

**Review by paloobi — CHANGES_REQUESTED** [review]
Also - in your tests, you need more setup in `beforeEach` - for instance, `createTables` really should only happen in `beforeEach`

File: __tests__/costumes.test.js (hunk 2 above)
Comment (paloobi) [inline __tests__/costumes.test.js]: prefer to manually check the expected properties
```suggestion
        expect(buttlessChaps.name).toBe("...");
        expect(buttleChaps.price).toBe(XYZ);
```

File: __tests__/costumes.test.js (hunk 1 above)
Comment (dyazdani) [reply __tests__/costumes.test.js]: Ok, I will do that.

File: utilities.js
Comment (dyazdani) [reply utilities.js:1]: Can I put the actual tests in a folder under `__tests__` so that I can run that folder when I run tests? Otherwise Jest tries to run `utilities.js` and it logs it as a failed test.

File: costumes.js (hunk 3 above)
Comment (dyazdani) [reply costumes.js]: I guess I don't really know when a null value would come up instead of undefined, so that seems reasonable. I did `undefined` because that is what the value was if the id passed into the function was wrong. The other reason was I thought being more specific with the data type versus using truthy or nullish would avoid a possible bug down the road if some code did ever cause the value being assessed to be null or an empty string, etc. for whatever reason. Perhaps that is premature optimization?

File: costumes.js (hunk 5 above)
Comment (dyazdani) [reply costumes.js:94]: Right you are! This error is thrown on its own: `Error: Could not delete row because id provided (3) does not exist in table.`

Comment (dyazdani) [reply costumes.js]: It does nothing and no error is thrown. Does this if statement look OK to you knowing that? It's worked as intended in tests to my knowledge.

**Comment by dyazdani** [comment]
@paloobi What exactly do you mean when you say "all the tests co-dependent on one function"? Can you provide an example of what you mean? Do you mean all the tests using `createTable`? Perhaps it's related to my previous question on the main PR thread (not sure if you saw it) about whether or not it is OK to use other adapters in an adapter test. Like say for example, use the `createCosutmer` adapter to populate the database for a test of `createOrder` adapter instead of composing a new query every time.

File: customers.js (hunk 7 above)
Comment (dyazdani) [reply customers.js]: Did it!

File: index.js (hunk 11 above)
Comment (dyazdani) [reply index.js:55]: Noted!

File: ordersCostumes.js (hunk 15 above)
Comment (dyazdani) [reply ordersCostumes.js]: It throws an error if the wrong ID is passed because of the foreign key constraint. It does not throw an error if no IDs are provided so I added the `NOT NULL` constraint as you suggested.

File: ordersCostumes.js (hunk 17 above)
Comment (dyazdani) [reply ordersCostumes.js:28]: Thought it might be useful for costume shop business's analytics. For example, you could see what other costumes were in an order that had a specific costume in it, and extrapolate that "people who buy _x_ also buy _y_". Perhaps a feature that is not essential at this juncture.

File: ordersCostumes.js (hunk 16 above)
Comment (dyazdani) [reply ordersCostumes.js]: OK. I'll remove it for now.

File: ordersCostumes.js (hunk 17 above)
Comment (dyazdani) [reply ordersCostumes.js:28]: I was trying to come up with more adapters the might be useful in hypothetical scenarios versus ones that would have a function right now in the project as is.

File: ordersCostumes.js (hunk 18 above)
Comment (dyazdani) [reply ordersCostumes.js]: I'll remove it.

File: ordersCostumes.js (hunk 20 above)
Comment (dyazdani) [reply ordersCostumes.js]: I'll remove it for now and do it in the web server layer like the others

File: utilities.js (hunk 23 above)
Comment (dyazdani) [reply utilities.js]: I would use `toStrictEqual` but the object keys are different in the database versus `utilities.js` (camelCase versus snake_case). Plus the time stamp gets converted to UTC in the database so it is a slightly different time stamp than the input date when I query for it. I could change the time zone for my DB to Pacific, but I didn't know if that was the right thing to do. I should have asked. Is that the right thing to do?

File: __tests__/costumes.test.js (hunk 2 above)
Comment (dyazdani) [reply __tests__/costumes.test.js]: Is that because this actually tests the value of each property instead of just matching the object properties?

File: utilities.js (hunk 22 above)
Comment (dyazdani) [reply utilities.js]: Done

File: seed/seedData.json (hunk 21 above)
Comment (dyazdani) [reply seed/seedData.json:49]: I added the existing costumes to the existing orders to the orders_costumes table.

File: utilities.js
Comment (paloobi) [reply utilities.js:1]: Hm, I didn't realize Jest would run everything in `__tests__` as if it were a test, I thought it only looked for files ending in .spec.js or .test.js

Try doing /utils/index.js and see if it works? If that doesn't work, add to the jest config:

```
modulePathIgnorePatterns: ["utils"]
```

File: costumes.js (hunk 3 above)
Comment (paloobi) [reply costumes.js]: In this case, since costume should be an object, any false-y value would represent an issue - I think it would probably actually be undefined, like you said, but this is a case where a truthy check is very common (same as in React, where we often get things from backend and will check if they are truthy to check if they are a defined object or not)

File: costumes.js (hunk 5 above)
Comment (paloobi) [reply costumes.js]: https://www.postgresql.org/docs/current/sql-delete.html

> The optional RETURNING clause causes DELETE to compute and return value(s) based on each row actually deleted. 

I think this is the best course of action. Do RETURNING at the end, and then just check if the returned costume was null

By the way, since it requires a RETURNING to return something, I'm not sure how this error was working and not breaking tests? Wouldn't it always have length 0 if there's no RETURNING clause? Maybe it returns the object by default?

File: ordersCostumes.js (hunk 17 above)
Comment (paloobi) [reply ordersCostumes.js:28]: That makes sense to me! Keep it, maybe add a comment about possible use-case so you can remember later

File: utilities.js (hunk 23 above)
Comment (paloobi) [reply utilities.js]: I think you're right to keep timestamp on DB as UTC. Look into how to get the timezone to change so your tests can match.

File: __tests__/costumes.test.js (hunk 2 above)
Comment (paloobi) [reply __tests__/costumes.test.js]: Let me put it this way:

`expect(buttlessChaps.name).toEqual("Some test name")` is a perfectly adequate test case, and will test both that the key is correct/not undefined AND the value is as expected

**Comment by paloobi** [comment]
> @paloobi What exactly do you mean when you say "all the tests co-dependent on one function"? Can you provide an example of what you mean? Do you mean all the tests using createTable? Perhaps it's related to my previous question on the main PR thread (not sure if you saw it) about whether or not it is OK to use other adapters in an adapter test. Like say for example, use the createCosutmer adapter to populate the database for a test of createOrder adapter instead of composing a new query every time.

I mean that every single test in your whole suite will fail if a bug is introduced in `matchesDatabase`. This will ultimately make your tests more flaky. I would recommend doing more direct checks for the values after each function is called.


I do think it's fine to use adapters in other tests, eg.

```
test('updateOrder updates the order', async () => {
    const todo = await createTodo("Test todo", "Lorem ipsum", true);

    const newName = "Test update";
    const newDescription =  "updated description";

    const udpatedTodo = await updateTodo(todo.id, newName, newDescription, false);

    expect(updatedTodo.name).toEqual(newName);
    expect(updatedTodo.description).toEqual(newDescription);
    expect(updatedTodo.isActive).toStrictEqual(true);
});
```

File: __tests__/utilities.js [hunk 24]
```diff
@@ -0,0 +1,396 @@
+// *** FOR JEST TESTING ***
+
+// Helper function for matching input objects with objects returned from DB queries
+const matchesDatabase = (inputObject, databaseObject) => {
```
Comment (AnthonyPinto) [inline __tests__/utilities.js]: So, I know Allie touched on this in an earlier comment, but I'm going to make a suggestion:
So, it isn't *always* a problem to have a util function that you use a lot, but in this case it is a tricky enough function that it would probably be reasonable to write a test for the helper function. There are circumstances that call for that kind of thing, but it should raise a yellow flag that you may be doing more than is needed.

In this case, the heart of the problem, is that you are testing more than you need to, or specifically you are testing the same thing repeatedly.

It would be reasonable to have a single test per table, that makes sure that a row in that table stores and returns all of the values that we expect it to. but this should only be needed in a single test per table - and I would argue it would be better to write each of those tests out explicitly. Maybe with a couple smaller helpers just for date conversions.

This is because the thing that these tests should be checking for is that you have created the table correctly, and they protect against a future change that breaks the function you use for creating tables - HOWEVER, once you have confirmed that your function created the table as expected, you should stop testing that the table is correctly formatted. that is testing postgres instead of your own code.

All your other tests should just check the specific value or values that are relevant to them.

The problems with using the current approach are 1. this function is complicated enough to contain it's own bugs or get new ones if refactored 2. testing everything everywhere actually makes it harder to realize what is broken when it breaks. Because lets say the bug is that your table is missing a column - you would want just the "table has expected columns" test to fail instead of every single test.

So those would be my suggestions. That being said, I could imagine debating this with a coworker because there is an argument to be made for having this kind of DRY all-in-one test function that can be run at least once per table.

So if you want to keep this function as is, I would say write a test for it (with dummy input and outputs) and try to limit it's usage to one or two cases per table where it feels appropriate.

File: utilities.js
Comment (dyazdani) [reply utilities.js:1]: Will do

File: costumes.js (hunk 3 above)
Comment (dyazdani) [reply costumes.js]: Makes perfect sense. Thank you for clarifying!

File: costumes.js (hunk 5 above)
Comment (dyazdani) [reply costumes.js]: Without a RETURNING clause, it was returning an array of objects representing the rows deleted. If nothing was deleted, it returned an empty array.

File: __tests__/costumes.test.js (hunk 2 above)
Comment (dyazdani) [reply __tests__/costumes.test.js]: Thank you for explaining that. Makes total sense!

File: __tests__/utilities.js (hunk 24 above)
Comment (dyazdani) [reply __tests__/utilities.js]: Thank you for the feedback, Anthony. Before I read this, I had already begun the process of taking `matchesDatabase` out of my tests. So, I'll just keep doing that, but you make a lot of very good points that I will keep in mind in the future.

File: __tests__/tests/costumes.test.js [hunk 25]
```diff
@@ -0,0 +1,325 @@
...
+
+    it("should throw an error if argument does not follow VARCHAR length constraint", async () => {
+        expect.hasAssertions();
+        try {
+            await createCostume(pool, getGownWithLongSize())
+        } catch (e) {
+            expect(e.name).toMatch('error');
+        }
+    })
+})
```
Comment (AnthonyPinto) [inline __tests__/tests/costumes.test.js]: I would actually argue against this category of tests

If you consider what it is that is being tested here, it boils down to whether or not you set the correct constraints when you created the tables (because the only other thing being tested is whether postgres is working as expected)

It's a bit on the line, but in most cases, I would say don't write a test that just check a config, and in this case, how you create the tables feels more like a config than like business logic. In other-words test code, not which options you selected.

Comment (dyazdani) [reply __tests__/tests/costumes.test.js]: Makes sense. How does one test if the configurations are working? Or do you?

File: costumes.js (hunk 5 above)
Comment (dyazdani) [reply costumes.js]: Just checked and it does the same thing with a RETURNING clause on DELETE. I changed my code to use destructuring assignment for the element that is in the array returned by the DELETE query, that way I can do a truthy check instead of checking the length property of the array. With that setup, it returns `undefined` if nothing was deleted.

**Review by paloobi — APPROVED** [review]
I like these new tests, they are much more readable. I don't have time to read through each one, so I am trusting you here. If they are all passing, I am happy. :)

File: __tests__/tests/costumes.test.js (hunk 25 above)
Comment (AnthonyPinto) [reply __tests__/tests/costumes.test.js]: Honestly, often you wouldn't write tests specifically for that. Typically you can think of the configs almost like implementation details that should be caught by the combination of testing the features that actually use them - and end-to-end smoke-tests that confirm that nothing major is busted.
